apps/aechannel/src/aesc_session_noise.erl

```diff
@@ -135,152 +167,24 @@ patterns() ->
 
 record_fields(st) -> record_info(fields, st);
 record_fields(_ ) -> no.
-%% ==================================================================
-
-%% Connection establishment
-
-connect(Host, Port, Opts) ->
-    start_link(#{fsm => self(), op => {connect, Host, Port, Opts}}).
 
-accept(#{ port := Port, responder := R } = Opts0, NoiseOpts) ->
-    TcpOpts = tcp_opts(listen, NoiseOpts),
-    lager:debug("listen: Opts0 = ~p; TcpOpts = ~p", [Opts0, TcpOpts]),
-    {ok, LSock} = aesc_listeners:listen(Port, R, TcpOpts),
-    Opts = Opts0#{lsock => LSock},
-    accept_(Opts, NoiseOpts).
-
-accept_(#{ reestablish := true
-         , chain_hash  := ChainH
-         , channel_id  := ChId
-         , responder   := R
-         , port        := Port } = Opts, NoiseOpts) ->
-    Regkey = responder_reestabl_regkey(ChainH, ChId, R, Port),
-    lager:debug("Regkey = ~p", [Regkey]),
-    gproc:reg(Regkey),
-    {ok, _Pid} = aesc_sessions_sup:start_child(
-                   [#{fsm => self(), op => {accept, Opts, NoiseOpts}}]);
-accept_(#{ initiator := I, responder := R, port := Port } = Opts, NoiseOpts) ->
-    Regkey = responder_regkey(R, I, Port),
-    lager:debug("Regkey = ~p", [Regkey]),
-    gproc:reg(Regkey),
-    aesc_sessions_sup:start_child([#{fsm => self(), op => {accept, Opts, NoiseOpts}}]).
+%% ==================================================================
+%% gen_server API
 
-start_link(Arg) when is_map(Arg) ->
-    %% Res = gen_server:start_link(?MODULE, Arg#{parent => self()}, ?GEN_SERVER_OPTS),
-    %% We don't use gen_server:start_link/3, since we want to use gen_server:enter_loop/3
-    %% in the init/1 function, and we want to avoid redundant `init_ack()` messages
-    %% sent by the `gen_server:init_it/6 function (a different kind of support has been
-    %% introduced in later versions of OTP.)
-    Res = proc_lib:start_link(?MODULE, init, [Arg#{parent => self()}]),
```

Comment:
I moved from this setup to using straight up `gen_server` with the connection establishing logic being called after the process has finished its `init` routine.